plugins/core/db/index.js

```diff
@@ -1,11 +1,63 @@
-const nano = require('nano');
-const Boom = require('boom');
+const nano = require("nano");
+const Boom = require("boom");
+
+function getFilters(queryParameters) {
+  let filters = [];
+  for (let parameter in queryParameters) {
+    if (
```

Comment:
I'd prefer an implementation like this:

```
filters = Object.keys(queryParameters)
  .map(parameterName => {
    if (parameterName === "searchString") {
       const filter = ....
       return filter;
    }
    if (other special case) {
      const filter = ...
      return filter;
    }
    // all the other special cases
    // the default case
  })
```

this results in smaller if blocks. returning early makes it much easier to read imo as there is no need for `else`.
the early returns also help to make it very clear that there is no other path the code could take later on in the function.